synapse/notifier.py

```diff
@@ -93,6 +93,13 @@ def notify(self, notifier, events, start_token, end_token):
                 self.appservice, set()
             ).discard(self)
 
+        # Cancel the timeout for this notifer if one exists.
+        if self.timer is not None:
+            try:
+                notifier.clock.cancel_call_later(self.timer)
+            except:
+                logger.exception("Failed to cancel notifier timer")
```

Comment:
Hmm, looks like cancel_call_later just calls IDelayedCall.cancel. Which as you point out will throw if the timer has already fired.